src/infinitemps.jl

```diff
@@ -69,8 +69,10 @@ function Base.getindex(l::InfiniteITensorSum, n1n2::Tuple{Int,Int})
   @assert n2 == n1 + 1
   return l.data[n1]
 end
-nsites(h::InfiniteITensorSum) = length(l.data)
-
+nsites(h::InfiniteITensorSum) = length(h.data)
+#Gives the range of the Hamiltonian. Useful for better optimized contraction in VUMPS
+nranges(h::InfiniteITensorSum) = order.(h.data)÷2
+nrange(h::InfiniteITensorSum, n::Int64) = order(h.data[n])÷2
```

Comment:
I guess this points to `nsites` being an overloaded term, since `nsites` would also be a good name for this. In fact, I don't I like `nrange` for this since the there could be a two-site interaction with a long range. Maybe `nsite_support` would be a better name for this.

Reply:
Though I may be misinterpreting what this function is supposed to do, in a general case.

Reply:
Fair point. I implicitly assumed that local hamiltonians are always involves contiguous sets of sites, it also makes part of the coding simpler. I can easily patch by checking indexes when contracting (due to the priming).
I actually tried to have something a bit more generic than evaluating the order, but I was struggling a bit with reading the SmallStrings, and therefore postponed it.
I think it is important to well-differentiate between the nsites of the MPS (the period of the motif), the support of the Hamiltonian and the range.
Happy to take any convention you judge better.

Reply:
Based on the proposal in #31 of using an `InfiniteMPOSum` representation, the MPOs could be constrained to be contiguous, in which case the range and support would be the same, but I think it is still useful to distinguish those two concepts.

For example, we should probably support allowing someone to input a long-range interacting Hamiltonian with terms that have support on two sites in terms of a 4-index tensor with non-contiguous sites (which we might then convert into an `InfiniteMPOSum` with contiguous sites equal to the range of the interaction).

Here is a code for finding the support of an operator with potentially non-contiguous sites:
```julia
using ITensors
using ITensorInfiniteMPS

ITensorInfiniteMPS.getcell(i::Index) = ITensorInfiniteMPS.getcell(tags(i))
function ITensors.findfirstsiteind(ψ::InfiniteMPS, i::Index)
  c = ITensorInfiniteMPS.getcell(i)
  i1 = translatecell(i, -(c - 1))
  n1 = findfirst(hascommoninds(i1), ψ[Cell(1)])
  return (c - 1) * nsites(ψ) + n1
end
function ITensors.findsites(ψ::InfiniteMPS, T::ITensor)
  s = filterinds(T; plev=0)
  return [ITensors.findfirstsiteind(ψ, i) for i in s]
end
ITensors.findsites(ψ::InfiniteCanonicalMPS, T::ITensor) = findsites(ψ.AL, T)

N = 2
heisenberg_space_shift(q̃nf, q̃sz) = [QN("Sz", 1 - q̃sz) => 1, QN("Sz", -1 - q̃sz) => 1]
heisenberg_space = fill(heisenberg_space_shift(1, 0), N)
s = infsiteinds("S=1/2", N; space=heisenberg_space)
initstate(n) = isodd(n) ? "↑" : "↓"
ψ = InfMPS(s, initstate)
model = Model("heisenberg")
H = InfiniteITensorSum(model, s)

@show findsites(ψ, H[(1, 2)])
@show findsites(ψ, H[(2, 3)])
@show findsites(ψ, H[(3, 4)])
@show findsites(ψ, H[(4, 5)])
```
This is following analogous functionality in `ITensors.jl` for finite MPS.

Reply:
Even for an MPO representation of the local Hamiltonian, we do not need the MPO to have contiguous sites (putting nothing and waiting for contraction is equivalent to putting identity matrices in the MPO).

Thank you for the findsites template, I updated the PR, and added some matching functionalities for other situations.

Reply:
Right, the MPO does not need to be contiguous. Perhaps it makes it simpler to make it contiguous for coding (in terms of prime level logic) in which case you can simply write a function that puts identity tensors between non-contiguous sites of the MPO.